akka-http-core/src/test/scala/akka/http/scaladsl/ClientSpec.scala

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.scaladsl
+
+import java.util.concurrent.ConcurrentHashMap
+
+import akka.actor.ActorSystem
+import akka.http.impl.engine.client.PoolGateway
+import akka.http.impl.settings.HostConnectionPoolSetup
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.HttpMethods._
+import akka.stream.ActorMaterializer
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.{ Matchers, PrivateMethodTester, WordSpec }
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future }
+
+class ClientSpec extends WordSpec with Matchers with PrivateMethodTester {
+  val testConf: Config = ConfigFactory.parseString("""
+    akka.loggers = ["akka.testkit.TestEventListener"]
+    akka.loglevel = ERROR
+    akka.stdout-loglevel = ERROR
+    windows-connection-abort-workaround-enabled = auto
+    akka.log-dead-letters = OFF
+    akka.http.server.request-timeout = infinite""")
+  implicit val system = ActorSystem(getClass.getSimpleName, testConf)
+  import system.dispatcher
+  implicit val materializer = ActorMaterializer()
+
+  "HTTP Client" should {
+
+    "reuse connection pool" in {
+      val hostPoolCache = PrivateMethod[ConcurrentHashMap[HostConnectionPoolSetup, Future[PoolGateway]]]('hostPoolCache)
+
+      val (_, hostname, port) = TestUtils.temporaryServerHostnameAndPort()
+      val bindingFuture = Http().bindAndHandleSync(_ ⇒ HttpResponse(), hostname, port)
+      val binding = Await.result(bindingFuture, 3.seconds)
+
+      val respFuture = Http().singleRequest(HttpRequest(POST, s"http://$hostname:$port/"))
+      val resp = Await.result(respFuture, 3.seconds)
+      resp.status shouldBe StatusCodes.OK
+
+      (Http() invokePrivate hostPoolCache()).size shouldBe 1
+
+      val respFuture2 = Http().singleRequest(HttpRequest(POST, s"http://$hostname:$port/"))
+      val resp2 = Await.result(respFuture, 3.seconds)
+      resp2.status shouldBe StatusCodes.OK
+
+      (Http() invokePrivate hostPoolCache()).size shouldBe 1
```

Comment:
I'm not a huge fan of invokePrivate, would rather make it `private[http]`. Anyone have strong opinions on it?